### PR TITLE
164 selection border affect layout 2

### LINKED ISSUE #164: Avoid affecting layout with selection border
Find a way for to avoid affecting the editor site layout when a component is selected.

It may be necessary to use an absolute positioned overlay, instead of adding the border directly to the component.

### PR DESCRIPTION
**Pull request by sampullman**
Close #164 

Replaces #213

### CODE REVIEW

**Comment by sampullman** [comment]
@abemscac I did my best to fix it, but probably broke some other things. I'll leave some comments in the code and we can try to review it on Monday.

File: libs/frontend/ui-build/src/components/BuildDndOverlay.vue [hunk 1]
```diff
@@ -79,35 +78,163 @@ const overlayStyle = computed(() => {
...
+  }
+}
+
+watch(
+  () => ({
+    selectedComponent: editor.value?.selectedComponent,
+    // Re-compute selection overlay width & height when active breakpoint changes.
+    activeBreakpoint: activeBreakpoint.value,
+    // Recompute on window resize
+    buildContentWindowSize: runtimeContext.buildContentWindowSize.value,
```
Comment (sampullman) [inline libs/frontend/ui-build/src/components/BuildDndOverlay.vue:167]: Also reset selection when window size changes.

File: libs/frontend/ui-build/src/components/BuildDndOverlay.vue [hunk 2]
```diff
@@ -79,35 +78,163 @@ const overlayStyle = computed(() => {
...
 })
+
+// Use `ref`+`watch` instead of `computed` because we have to wait for the screen to (re)render
+// before calculating the width and height of the component on page load & after selection.
+const selectionOverlayStyle = ref()
+
+const computeSelectionOverlay = async (selectedComponent: IComponent | undefined) => {
+  if (selectedComponent) {
+    // This sleep is necessary for buildContentWindowInner and componentElement
+    await sleep(1)
```
Comment (sampullman) [inline libs/frontend/ui-build/src/components/BuildDndOverlay.vue:116]: Seems to be more effective than `nextTick`, though I'm not sure why.

File: libs/frontend/ui-build/src/components/BuildDndOverlay.vue [hunk 3]
```diff
@@ -79,35 +78,163 @@ const overlayStyle = computed(() => {
...
+      const { top, left, width, height } = normalizeDimensions(
+        elementRelativeTop,
+        elementRelativeLeft,
+        elementWidth,
+        elementHeight,
+        buildContentWindowInner,
+      )
+
+      selectionOverlayStyle.value = {
+        top: `${top - 1}px`,
```
Comment (sampullman) [inline libs/frontend/ui-build/src/components/BuildDndOverlay.vue:149]: This `-1` appears to position the border correctly in all cases, but I don't know why it's necessary. I also increased the border width to 2px.

File: libs/frontend/feature-render-builder/src/lib/dnd/builder-dnd-component.ts [hunk 4]
```diff
@@ -50,24 +63,8 @@ const renderImageHover = (
     top = imgComputedStyle.top
     right = imgComputedStyle.right
 
-    if (position === 'absolute' && buildContentWindowInner) {
```
Comment (sampullman) [inline libs/frontend/feature-render-builder/src/lib/dnd/builder-dnd-component.ts:53]: Removing this probably broke something, but it seemed to fix selecting/hovering an absolute positioned image. One remaining issue is hovering over the edge of an absolute image/svg causes it to change position in some situations.

**Comment by abemscac** [comment]
Got it, I'll try to understand the code in this PR today.
Thank you so much! 🙏

**Pull request merged**